### DISCUSSION TITLE
Is it possible to use a WebGL texture as an image.

### DISCUSSION THREAD

**Discussion by angad-k** [emilk/egui#1626] [Q&A]
Hey there! I am working on a rust application that uses webgl calls through the websys crate to render 3d onto the screen. I have been trying since yesterday to integrate egui/eframe in the app but am having problems with displaying the 3d scene in egui. I am rendering the 3d scene to an offscreen texture and want to use it in egui as an image - however egui takes in a `TextureId` while I have a `WebGlTexture` which I want to use. Any way I could go about doing this?

**Comment by KentaTheBugMaker** [comment]
first you should reed examples/custom_3d_glow or examples/custom_3d_three.
You should use glow functions by following reasons .
 * web_sys::WebGLxxxx is not supported directly but we support it via glow.
 * glow naming is much better. 
 * you can access glow resources via eframe. 
 
displaying textured rectangle instead of triangle in examples.

**Reply by angad-k** [reply]
What about `egui_web`? It appears to be discontinued now but it seems that it was written using plain webgl in Rust. Would that be useful for my case? - or is using glow the only way to go?

**Reply by KentaTheBugMaker** [reply]
egui_web is almost used as eframe backend due to complexity.
you can same job as webgl .
but maybe restrict js related jobs. 
## glow painter history on web.
|status|experimental | default |only one | end of egui_web|
|------|-------------|-------|---------| -------------------|
|date| 2021/11/4  | 2021/12/31 |2022/3/12|2022/5/1|
|commit number|#868|#1020|#1356 | #1545|
## why old webgl1/webgl2 dropped.
removed egui_web's painter code is almost same as glow painter  and backported from glow painter .
## alternative solution.
some day wgpu will preferred.
